lld/ELF/SyntheticSections.cpp

```diff
@@ -655,6 +655,10 @@ void GotSection::addConstant(const Relocation &r) { relocations.push_back(r); }
 void GotSection::addEntry(const Symbol &sym) {
   assert(sym.auxIdx == symAux.size() - 1);
   symAux.back().gotIdx = numEntries++;
+  if (sym.hasFlag(NEEDS_GOT_AUTH)) {
+    assert(config->emachine == EM_AARCH64);
```

Comment:
This assert and the above assert are not useful.

Reply:
I've deleted the assert, thanks, see dca427bbf8ab6d53fd01b3d1799d4d723992edd2.

Regarding the assertion above - I'll delete that and similar ones in `addTlsDescEntry` and `addDynTlsEntry` if they are considered useless (I just don't want to mess new features with refactoring existing code).

Reply:
Update. Opened #98028 deleting similar assertions in

- `GotSection::addEntry`
- `GotSection::addTlsDescEntry`
- `GotSection::addDynTlsEntry`

Reply:
`assert(sym.auxIdx == symAux.size() - 1);` assertions are useful to prevent allocating more than one entries for one symbol. I used the assertions to prevent some failures. `NEEDS_GOT_AUTH` is clearly AArch64 specific.

Reply:
I probably misunderstood you phrase "and the **above assert** are not useful" - I thought you mean the one I deleted. Thanks for explaining your point, now I get what you meant.

The aarch64-specific useless assert was deleted, see dca427bbf8ab6d53fd01b3d1799d4d723992edd2